services/web/server/tests/unit/with_dbs/07/test_users.py

```diff
@@ -112,6 +116,38 @@ async def fake_tokens(logged_user, tokens_db):
     return all_tokens
 
 
+# TESTS --------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "user_role,expected",
+    [
+        ("UserNotRegistered", False),
+        (UserRole.ANONYMOUS, False),
+        (UserRole.GUEST, True),
+        (UserRole.USER, False),
+        (UserRole.TESTER, False),
+    ],
+)
+async def test_is_user_guest(client: TestClient, user_role, expected: bool):
+
+    if user_role == "UserNotRegistered":
+        # WARNING! not even a user, so it cannot be a guest!
+        assert await is_user_guest(client.app, 123456789) == expected
```

Comment:
minor: if you parametrize a test and you need to add a ```if...else``` case just for one of the parameter I think it is clearer intent to create a separate test

Reply:
True